drivers/gpu/drm/panel/panel-shift-sh8804b.c

```diff
@@ -0,0 +1,1189 @@
+	connector->display_info.height_mm = mode->height_mm;
+	drm_mode_probed_add(connector, mode);
+
+	return 1;
+}
+
+static const struct drm_panel_funcs shift_sh8804b_panel_funcs = {
+	.prepare = shift_sh8804b_prepare,
+	.unprepare = shift_sh8804b_unprepare,
+	.get_modes = shift_sh8804b_get_modes,
+};
+
+
+static int shift_sh8804b_panel_bl_update_status(struct backlight_device *bl)
+{
+	struct mipi_dsi_device *dsi = bl_get_data(bl);
+	int err;
+	u16 brightness = (u16)backlight_get_brightness(bl);
+
+	err = mipi_dsi_dcs_set_display_brightness_large(dsi, brightness);
+	if (err < 0)
+		return err;
+
+	return 0;
+}
+
+static const struct backlight_ops shift_sh8804b_panel_bl_ops = {
+	.update_status = shift_sh8804b_panel_bl_update_status,
+};
+
+static struct backlight_device *
+shift_sh8804b_create_backlight(struct mipi_dsi_device *dsi)
+{
+	struct device *dev = &dsi->dev;
+	const struct backlight_properties props = {
+		.type = BACKLIGHT_PLATFORM,
+		.brightness = 512,
+		.max_brightness = 2047,
+	};
+
+	return devm_backlight_device_register(dev, dev_name(dev), dev, dsi,
+					      &shift_sh8804b_panel_bl_ops, &props);
+}
+
+static int shift_sh8804b_probe(struct mipi_dsi_device *dsi)
+{
+	struct device *dev = &dsi->dev;
+	struct shift_sh8804b *ctx;
+	int ret;
+
+	dev_info(dev, "Probing shift_sh8804b panel\n");
+
+	ctx = devm_kzalloc(dev, sizeof(*ctx), GFP_KERNEL);
+	if (!ctx)
+		return -ENOMEM;
+
+	ctx->supplies[0].supply = "vci";
+	ctx->supplies[1].supply = "vddio";
+	ret = devm_regulator_bulk_get(dev, ARRAY_SIZE(ctx->supplies),
+				      ctx->supplies);
+	if (ret < 0)
+		return dev_err_probe(dev, ret, "Failed to get regulators\n");
+
+	ctx->reset_gpio = devm_gpiod_get(dev, "reset", GPIOD_OUT_HIGH);
+	if (IS_ERR(ctx->reset_gpio))
+		return dev_err_probe(dev, PTR_ERR(ctx->reset_gpio),
+				     "Failed to get reset-gpios\n");
+
+	ctx->dsi = dsi;
+	mipi_dsi_set_drvdata(dsi, ctx);
+
+	dsi->lanes = 4;
+	dsi->format = MIPI_DSI_FMT_RGB888;
+
+	drm_panel_init(&ctx->panel, dev, &shift_sh8804b_panel_funcs,
+		       DRM_MODE_CONNECTOR_DSI);
+	ctx->panel.prepare_prev_first = true;
+
+	ctx->panel.backlight = shift_sh8804b_create_backlight(dsi);
+	if (IS_ERR(ctx->panel.backlight))
+		return dev_err_probe(dev, PTR_ERR(ctx->panel.backlight),
+				     "Failed to create backlight\n");
+
+	drm_panel_add(&ctx->panel);
+
+	ret = mipi_dsi_attach(dsi);
+	if (ret < 0) {
+		dev_err(dev, "Failed to attach to DSI host: %d\n", ret);
+		drm_panel_remove(&ctx->panel);
+		return ret;
+	}
+
+	dev_info(dev, "shift_sh8804b panel probed!\n");
+
+	return 0;
+}
+
+static void shift_sh8804b_remove(struct mipi_dsi_device *dsi)
+{
+	struct shift_sh8804b *ctx = mipi_dsi_get_drvdata(dsi);
+	int ret;
+
+	ret = mipi_dsi_detach(dsi);
+	if (ret < 0)
+		dev_err(&dsi->dev, "Failed to detach from DSI host: %d\n", ret);
+
+	drm_panel_remove(&ctx->panel);
+}
+
+static const struct of_device_id shift_sh8804b_of_match[] = {
+	{ .compatible = "shift,sh8804b" }, // FIXME
```

Comment:
Maybe you can find a nice model name or something from your display vendor?

E.g. for Fairphone 5 I have a pdf for the company Boe with the "Model Name" BF065GBM-TK0-7DP0 which is exactly the combo of driver IC (rm692e5) plus everything else that's (I assume) specific to Fairphone 5.

Otherwise probably upstream is happy enough if you make it something else, see e.g. Fairphone 5 panel upstream (to be clear, that doesn't follow what I was writing, see Fairphone 4 panel there: `panel-himax-hx83112a.c`)

Reply:
This would be `TA067FVMK01-08` then, also using `rm692e5`, but they also made adjustments specific for SP8.